clang/include/clang/Basic/DiagnosticDriverKinds.td

```diff
@@ -295,6 +295,9 @@ def err_drv_fsycl_with_c_type : Error<
 def warn_drv_sycl_offload_target_duplicate : Warning<
   "SYCL offloading target '%0' is similar to target '%1' already specified; "
   "will be ignored">, InGroup<SyclTarget>;
+def warn_drv_sycl_deprecated_triple_component : Warning<"'%0' component of '%1'"
```

Comment:
Clang diagnostics have some strange format and punctuation requirements (See: https://github.com/intel/llvm/pull/4243).

So this diagnostic should be changed to - 

`"'%0' component of '%1' SYCL offloading target triple is deprecated; effective triple is '%2'`

I'm also not sure about 'effective triple is'. Is there a reason we can't just say - use '%2'

Reply:
My suggestion is: `"'%0' component of '%1' SYCL offloading target triple is deprecated; using '%2' instead"`

Reply:
I used "effective triple" instead of "using" because currently the diagnostics prints "normalized" triple, which includes all "required" components: target, vendor and OS.
I don't want suggest users to set all these components. I'd like them to set only target component as we don't support vendor and OS components other than "unknown", which is set by the compiler automatically.
I can fix apply Aaron's suggestion + pass only target component of the triple as %2.
Does it sound okay?

Reply:
Sound good to me

Reply:
Sounds sensible to me! Btw, I went with `using` instead of `use` because it implies "this is what's being used instead of what you wrote" and it sounds to me like with @bader's suggested approach, we should go with `use` because that implies "use this other thing instead to not get this diagnostic in the future".

Reply:
Right, sorry about confusion. I realized that I misinterpret the suggestion after I posted the comment.
What would be the preference here: suggest users how to fix value of `-fsycl-targets` option or notify them that old value is deprecated now compiler implicitly use the triple w/o environment component?
The first option seems to be more helpful.

Reply:
> My suggestion is: `"'%0' component of '%1' SYCL offloading target triple is deprecated; using '%2' instead"`

I applied this suggestion.
Let me know if you think we should change the message from "FYI" to "recommendation".

Reply:
Sorry @bader, I just got back onto this review. I think the `use Blah instead` formulation is better for users, but I didn't know if that would be easy or hard for you to produce. If you can definitively tell the user a better choice, I'd say let's go with `use %2 instead` and if you can't definitively tell the user a better choice, we can go with `using %2 instead` to at least tell the user how we've interpreted their command line. WDYT?

Reply:
We can go with `use %2 instead`, but I just realized that it can be confusing.
For instace:
> 'sycldevice' component of 'spir64-unknown-unknown-sycldevice' SYCL offloading target triple is deprecated; use 'spir64' instead

The first part says only about one component of the triple, where as the second part references to the whole target triple.
Probably we should just say that the whole triple is deprecated w/o adding details about "environment" component.

> 'spir64-unknown-unknown-sycldevice' SYCL offloading target triple is deprecated; use 'spir64' instead

Reply:
I just found an existing warning we can re-use for that:

def warn_drv_deprecated_arg : Warning<
  "argument '%0' is deprecated, use '%1' instead">, InGroup<Deprecated>;

Reply:
Yeah, let's go with that warning. Thanks, @bader!

Reply:
Done with https://github.com/intel/llvm/pull/3929/commits/339849728997f5fb29688598483a86946aa9547a.